pandas/core/base.py

```diff
@@ -736,28 +735,15 @@ def argmax(
         nv.validate_minmax_axis(axis)
         skipna = nv.validate_argmax_with_skipna(skipna, args, kwargs)
 
+        if skipna and len(delegate) > 0 and isna(delegate).all():
+            raise ValueError("Encountered all NA values")
+        elif not skipna and isna(delegate).any():
+            raise ValueError("Encountered an NA value with skipna=False")
+
```

Comment:
huh IIRC the idea with the deprecation was that eventually we'd be able to make this method just `return self.array.argmax(skipna=skipna)`

Reply:
i might be confusing this with argsort; there is a TODO(3.0) comment in Series.argsort about using self.array.argsort directly